extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -0,0 +1,259 @@
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.jetbrains.annotations.NotNull;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class IcebergCatalogAdapter {
+    private final Catalog catalog;
+    private final FileIO fileIO;
+    private final IcebergInstructions instructions;
+
+    private final WeakHashMap<PartitionAwareSourceTable, AbstractTableLocationProvider> tableLocationProviders;
+
+    /**
+     * Construct an IcebergCatalogAdapter given a set of configurable instructions.
+     */
+    @SuppressWarnings("unused")
+    IcebergCatalogAdapter(
+            @NotNull final Catalog catalog,
+            @NotNull final FileIO fileIO) {
+        this(catalog, fileIO, IcebergInstructions.builder().build());
+    }
+
+    /**
+     * Construct an IcebergCatalogAdapter given a set of configurable instructions.
+     */
+    IcebergCatalogAdapter(
+            @NotNull final Catalog catalog,
+            @NotNull final FileIO fileIO,
+            @NotNull final IcebergInstructions instructions) {
+        this.catalog = catalog;
+        this.fileIO = fileIO;
+        this.instructions = instructions;
+        tableLocationProviders = new WeakHashMap<>();
+    }
+
+    static TableDefinition fromSchema(
+            final Schema schema,
+            final PartitionSpec partitionSpec,
+            final IcebergInstructions instructions) {
+        final Map<String, String> renameMap = instructions.columnRenameMap();
+
+        final Set<String> partitionNames =
+                partitionSpec.fields().stream()
+                        .map(PartitionField::name)
+                        .map(colName -> renameMap.getOrDefault(colName, colName))
+                        .collect(Collectors.toSet());
+
+        final List<ColumnDefinition<?>> columns = new ArrayList<>();
+
+        for (final Types.NestedField field : schema.columns()) {
+            final String name = renameMap.getOrDefault(field.name(), field.name());
+            final Type type = field.type();
+            final io.deephaven.qst.type.Type<?> qstType = convertPrimitiveType(type);
+            final ColumnDefinition<?> column;
+            if (partitionNames.contains(name)) {
+                column = ColumnDefinition.of(name, qstType).withPartitioning();
+            } else {
+                column = ColumnDefinition.of(name, qstType);
+            }
+            columns.add(column);
+        }
+
+        return TableDefinition.of(columns);
+    }
+
+    static io.deephaven.qst.type.Type<?> convertPrimitiveType(final Type icebergType) {
+        final Type.TypeID typeId = icebergType.typeId();
+        switch (typeId) {
+            case BOOLEAN:
+                return io.deephaven.qst.type.Type.booleanType();
```

Comment:
I wonder about nullability. We do *not* let users put unboxed `Booleans` in `Tables`.

Reply:
Is it sufficient to use:
```
                return io.deephaven.qst.type.Type.booleanType().boxedType();
```
Does that work, or do you run into some bug when you used it with real data?